polar-core/src/error.rs

```diff
@@ -498,3 +519,7 @@ impl fmt::Display for ValidationError {
         }
     }
 }
+
+pub fn invalid_state_error<A>(msg: String) -> Result<A, RuntimeError> {
+    Err(RuntimeError::InvalidState { msg })
+}
```

Comment:
Why's this one a top-level function and `unsupported()` is in the `impl RuntimeError` block?